ci-automation/vendor-testing/azure.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+# Copyright (c) 2021 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+set -euo pipefail
+
+set -x
+
+# Test execution script for the qemu vendor image.
+# This script is supposed to run in the SDK container.
+
+# $@ now contains tests / test patterns to run
+
+source ci-automation/vendor_test.sh
+
+board="${CIA_ARCH}-usr"
+basename="ci-${CIA_VERNUM//+/-}"
+azure_instance_type="${AZURE_MACHINE_SIZE}"
+
+testscript="$(basename "$0")"
+
+# Fetch the Azure image if not present
+if [ -f "${AZURE_IMAGE_NAME}" ] ; then
+    echo "++++ ${testscript}: Using existing ${work_dir}/${AZURE_IMAGE_NAME} for testing ${CIA_VERNUM} (${CIA_ARCH}) ++++"
+else
+    echo "++++ ${testscript}: downloading ${AZURE_IMAGE_NAME} for ${CIA_VERNUM} (${CIA_ARCH}) ++++"
+    copy_from_buildcache "images/${CIA_ARCH}/${CIA_VERNUM}/${AZURE_IMAGE_NAME}.bz2" .
+    lbunzip2 "${AZURE_IMAGE_NAME}.bz2"
+fi
+
+if [[ "${CIA_ARCH}" == "arm64" ]]; then
+  AZURE_USE_GALLERY="--azure-use-gallery"
+fi
+
+set -o noglob
+
+run_kola_tests() {
+    local instance_type="${1}"; shift
+    local instance_tapfile="${1}"; shift
+
+    timeout --signal=SIGQUIT 20h \
+    kola run \
+    --board="${board}" \
+    --basename="${basename}" \
+    --parallel="${AZURE_PARALLEL}" \
+    --offering=basic \
+    --platform=azure \
+    --azure-image-file="${AZURE_IMAGE_NAME}" \
+    --azure-location="${AZURE_LOCATION}" \
+    --azure-profile="${AZURE_PROFILE}" \
+    --azure-auth="${AZURE_AUTH_CREDENTIALS}" \
+    --torcx-manifest=${CIA_TORCX_MANIFEST} \
+    --tapfile="${instance_tapfile}" \
+    --azure-size=${instance_type} \
+    ${AZURE_USE_GALLERY} \
+    ${AZURE_HYPER_V_GENERATION:+--azure-hyper-v-generation=${AZURE_HYPER_V_GENERATION}} \
```

Comment:
you can look at the gce test where "gvnic" as instance type was used to set a kola parameter and we can do the same here for v1 vs v2